ONEM-9645 LIVE playback seek issues

Wlodzimierz Lipert <wlodzimierz.lipert@redembedded.com>

diff --git a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
index bad8747dcc5..c920ba5c27f 100644
--- a/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp
@@ -179,27 +179,33 @@ void MediaPlayerPrivateGStreamerMSE::seek(double time)
     if (UNLIKELY(!m_pipeline || m_errorOccured))
         return;
 
-    GST_INFO("[Seek] seek attempt to %f secs", time);
+    GST_INFO_OBJECT(m_pipeline.get(), "attempt to %f", time);
 
     // Avoid useless seeking.
-    float current = currentMediaTime().toFloat();
+    const auto current = currentMediaTime().toDouble();
     if (time == current) {
         if (!m_seeking)
             timeChanged();
         return;
     }
 
-    if (isLiveStream())
+    if (isLiveStream() && time > current)
+    {
+        GST_WARNING_OBJECT(
+            m_pipeline.get(),
+            "abort (future time %f) on live stream, curr time %f",
+            time, current);
         return;
+    }
 
     if (m_seeking && m_seekIsPending) {
         m_seekTime = time;
         return;
     }
 
-    GST_DEBUG("Seeking from %f to %f seconds", current, time);
+    GST_DEBUG_OBJECT(m_pipeline.get(), "from %f to %f", current, time);
 
-    float prevSeekTime = m_seekTime;
+    auto prevSeekTime = m_seekTime;
     m_seekTime = time;
 
     if (!doSeek()) {
@@ -209,7 +215,7 @@ void MediaPlayerPrivateGStreamerMSE::seek(double time)
     }
 
     m_isEndReached = false;
-    GST_DEBUG("m_seeking=%s, m_seekTime=%f", m_seeking ? "true" : "false", m_seekTime);
+    GST_DEBUG_OBJECT(m_pipeline.get(), "m_seeking=%s, m_seekTime=%f", m_seeking ? "true" : "false", m_seekTime);
 }
 
 void MediaPlayerPrivateGStreamerMSE::configurePlaySink()
